crates/bitwarden/src/tool/generators/username_forwarders/forwardemail.rs

```diff
@@ -0,0 +1,129 @@
+use base64::Engine;
+use reqwest::{header::CONTENT_TYPE, StatusCode};
+
+use crate::{
+    error::{Error, Result},
+    util::BASE64_ENGINE,
+};
+
+pub async fn generate(
+    http: &reqwest::Client,
+    api_token: String,
+    domain: String,
+    website: Option<String>,
+) -> Result<String> {
+    generate_with_api_url(
+        http,
+        api_token,
+        domain,
+        website,
+        "https://api.forwardemail.net".into(),
+    )
+    .await
+}
+
+async fn generate_with_api_url(
+    http: &reqwest::Client,
+    api_token: String,
+    domain: String,
+    website: Option<String>,
+    api_url: String,
+) -> Result<String> {
+    let api_token_b64 = BASE64_ENGINE.encode(format!("{api_token}:").as_bytes());
+
+    let description = super::format_description(&website);
+
+    #[derive(serde::Serialize)]
+    struct Request {
+        labels: Option<String>,
+        description: String,
+    }
+
+    let response = http
+        .post(format!("{api_url}/v1/domains/{domain}/aliases"))
+        .header(CONTENT_TYPE, "application/json")
+        .bearer_auth(api_token_b64)
+        .json(&Request {
+            description,
+            labels: website,
+        })
+        .send()
+        .await?;
+
+    if response.status() == StatusCode::UNAUTHORIZED {
+        return Err("Invalid Forward Email API key.".into());
+    }
+
+    // Throw any other errors
+    response.error_for_status_ref()?;
+
+    #[derive(serde::Deserialize)]
+    struct ResponseDomain {
+        name: Option<String>,
+    }
+    #[derive(serde::Deserialize)]
+    struct Response {
+        name: String,
+        domain: ResponseDomain,
+
+        message: Option<String>,
+        error: Option<String>,
+    }
+    let status = response.status();
+    let response: Response = response.json().await?;
+
+    if let Some(message) = response.message {
+        return Err(Error::ResponseContent { status, message });
+    }
```

Comment:
Is this correct? Seems weird to throw an error on `message`?

Reply:
The API response when there's an error is like this:
```
{
  "statusCode": 401,
  "error": "Unauthorized",
  "message": "Invalid API token."
}
```

So the `message` field is just an explanation of the `error` field. Not a great name but that's what the API gives us.